### ISSUE TITLE
[MPS] `sort` and `argsort` raise `RuntimeError` on empty inputs

### ISSUE THREAD

**Issue by hchau630** [pytorch/pytorch#117941]
### 🐛 Describe the bug

Calling sort and argsort on tensors with no elements results in RuntimeError.

```python
>>> import torch
>>> torch.zeros((0,), device='mps').sort()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: [srcBuf length] > 0 INTERNAL ASSERT FAILED at "/Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/native/mps/OperationUtils.mm":327, please report a bug to PyTorch. Placeholder tensor is empty!
```
This also happens with any shape where the tensor has zero elements, and also occurs with argsort.

### Versions

```
Collecting environment information...
PyTorch version: 2.1.2
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.5.1 (arm64)
GCC version: Could not collect
Clang version: 12.0.0 (clang-1200.0.32.28)
CMake version: version 3.26.0
Libc version: N/A

Python version: 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:35:25) [Clang 16.0.6 ] (64-bit runtime)
Python platform: macOS-13.5.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] flake8==6.1.0
[pip3] numpy==1.26.3
[pip3] torch==2.1.2
[pip3] torchdiffeq==0.2.3
[conda] numpy                     1.26.3                   pypi_0    pypi
[conda] torch                     2.1.2                    pypi_0    pypi
[conda] torchdiffeq               0.2.3                    pypi_0    pypi
```

cc @malfet @kulinseth @albanD @DenisVieriu97 @razarmehr

**Comment by malfet** [comment]
Actually, it's already fixed in latest nightly (and likely in 2.2 as well)

**Comment by malfet** [comment]
Duplicate of https://github.com/pytorch/pytorch/issues/107284
